src/query/exceptions.hpp

```diff
@@ -112,16 +112,45 @@ class QueryRuntimeException : public QueryException {
   using QueryException::QueryException;
 };
 
+enum class AbortReason : uint8_t {
+  NO_ABORT = 0,
+
+  // transaction has been requested to terminate, ie. "TERMINATE TRANSACTIONS ..."
+  TERMINATED = 1,
+
+  // server is gracefully shutting down
+  SHUTDOWN = 2,
+
+  // the transaction timeout has been reached. Either via "--query-execution-timeout-sec", or a per-transaction timeout
+  TIMEOUT = 3,
+};
+
 // This one is inherited from BasicException and will be treated as
 // TransientError, i. e. client will be encouraged to retry execution because it
 // could succeed if executed again.
 class HintedAbortError : public utils::BasicException {
  public:
   using utils::BasicException::BasicException;
-  HintedAbortError()
-      : utils::BasicException(
-            "Transaction was asked to abort either because it was executing longer than time specified or another user "
-            "asked it to abort.") {}
+  HintedAbortError(AbortReason reason) : utils::BasicException(AsMsg(reason)), reason_{reason} {}
```

Comment:
Clang is complaining to me here: "Single-argument constructors must be marked explicit to avoid unintentional implicit conversions". I am not sure if we should add explicit, seems fine this way too.

Reply:
I'll make it explicit...see no reason not to.

Reply:
fixed in latest commit